src/Widgets/WifiMenuItem.vala

```diff
@@ -173,6 +173,14 @@ public class Network.WifiMenuItem : Gtk.ListBoxRow {
             case NM.DeviceState.ACTIVATED:
                 connect_button_revealer.reveal_child = false;
                 break;
+            default:
+                /* Could be a lot of various status (for now, one of:
+                 * `DISCONNECTED', `IP_CHECK', `CONFIG', `UNKNOWN',
+                 * `UNAVAILABLE', `IP_CONFIG', `SECONDARIES', `UNMANAGED',
+                 * `NEED_AUTH', `DEACTIVATING'). Showing the spinner might let
+                 * end-user try to have a look to what is going on. */
+                spinner.active = true;
```

Comment:
This seems like we should handle this more carefully as well instead of just silencing the warning. It seems like a spinner isn't appropriate for states like "disconnected" or "unavailable" for example

Reply:
I’ll see what I can propose to better wrap all those cases.